### PR TITLE
improve test coverage

### LINKED ISSUE #41: Can't save catalog yaml representation
intake version 0.5.4
intake-stac version 0.2.2

Currently, opening a stac catalog (JSON) gives us an intake catalog (which can be represented as YAML). We inherit intake's `save` method, but that method fails with the following traceback
```
cat = intake.open_stac_catalog('https://storage.googleapis.com/pdd-stac/disasters/catalog.json')
cat.save('pdd-stac.yaml')
```

```pytb
---------------------------------------------------------------------------
ConstructorError                          Traceback (most recent call last)
<ipython-input-97-91d0ff8d57f2> in <module>
----> 1 cat.save('pdd-stac.yaml')
      2 
      3 ''' 
      4 ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object:satstac.item.Item'
      5   in "<unicode string>", line 4, column 17:

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/intake/catalog/base.py in save(self, url, storage_options)
    290         from fsspec import open_files
    291         with open_files([url], **(storage_options or {}), mode='wt')[0] as f:
--> 292             f.write(self.serialize())
    293 
    294     @reload_on_change

~/Documents/GitHub/intake-stac/intake_stac/catalog.py in serialize(self)
     74         output = {"metadata": self.metadata, "sources": {}}
     75         for key, entry in self.items():
---> 76             output["sources"][key] = yaml.safe_load(entry.yaml())["sources"]
     77         return yaml.dump(output)
     78 

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/__init__.py in safe_load(stream)
    160     to be safe for untrusted input.
    161     """
--> 162     return load(stream, SafeLoader)
    163 
    164 def safe_load_all(stream):

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/__init__.py in load(stream, Loader)
    112     loader = Loader(stream)
    113     try:
--> 114         return loader.get_single_data()
    115     finally:
    116         loader.dispose()

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in get_single_data(self)
     41         node = self.get_single_node()
     42         if node is not None:
---> 43             return self.construct_document(node)
     44         return None
     45 

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_document(self, node)
     50             self.state_generators = []
     51             for generator in state_generators:
---> 52                 for dummy in generator:
     53                     pass
     54         self.constructed_objects = {}

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_yaml_map(self, node)
    403         data = {}
    404         yield data
--> 405         value = self.construct_mapping(node)
    406         data.update(value)
    407 

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_mapping(self, node, deep)
    208         if isinstance(node, MappingNode):
    209             self.flatten_mapping(node)
--> 210         return super().construct_mapping(node, deep=deep)
    211 
    212     def construct_yaml_null(self, node):

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_mapping(self, node, deep)
    133                 raise ConstructorError("while constructing a mapping", node.start_mark,
    134                         "found unhashable key", key_node.start_mark)
--> 135             value = self.construct_object(value_node, deep=deep)
    136             mapping[key] = value
    137         return mapping

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_object(self, node, deep)
     90                     constructor = self.__class__.construct_mapping
     91         if tag_suffix is None:
---> 92             data = constructor(self, node)
     93         else:
     94             data = constructor(self, tag_suffix, node)

~/miniconda3/envs/intake-stac/lib/python3.8/site-packages/yaml/constructor.py in construct_undefined(self, node)
    417 
    418     def construct_undefined(self, node):
--> 419         raise ConstructorError(None, None,
    420                 "could not determine a constructor for the tag %r" % node.tag,
    421                 node.start_mark)

ConstructorError: could not determine a constructor for the tag 'tag:yaml.org,2002:python/object:satstac.item.Item'
  in "<unicode string>", line 4, column 17:
          stac_obj: !!python/object:satstac.item.Item
```

You can use the intake `print(cat['Houston-East-20170831-103f-100d-0f4f-RGB'].yaml())` method to see where these exclamation points are:
```
sources:
  Houston-East-20170831-103f-100d-0f4f-RGB:
    args:
      stac_obj: !!python/object:satstac.item.Item
        _assets_by_common_name: null
        _collection: null
        _data:
          assets:
            mosaic:
              href: https://storage.googleapis.com/pdd-stac/disasters/hurricane-harvey/0831/Houston-East-20170831-103f-100d-0f4f-3-band.tif
              title: 3 Band RGB Mosaic
              type: image/vnd.stac.geotiff; cloud-optimized=true
            thumbnail:
              href: https://storage.googleapis.com/pdd-stac/disasters/hurricane-harvey/0831/Houston-East-20170831-103f-100d-0f4f-3-band.png
              title: Thumbnail
              type: image/png
          bbox:
          - -95.73737276800716
          - 29.561332400220497
          - -95.05332428370095
          - 30.157560439570304
          geometry:
```

### PR DESCRIPTION
**Pull request by jhamman**
I'm trying to fix #41 before we merge this.

### CODE REVIEW

**Comment by scottyhq** [comment]
@jhamman - want to go ahead and remove the Travis CI that is failing https://github.com/intake/intake-stac/issues/51 and merge this? not sure what's up with Documentation build (i see ;logs' no longer available), so another commit might sort things

**Comment by jhamman** [comment]
@scottyhq - I removed the travis yaml in this PR now.

**Comment by andersy005** [comment]
By the way, now that Readthedocs provides automatic build of documentation for pull requests: https://docs.readthedocs.io/en/stable/guides/autobuild-docs-for-pull-requests.html, the Github action documentation build can be removed.

File: intake_stac/catalog.py [hunk 1]
```diff
@@ -5,9 +5,12 @@
 from intake.catalog import Catalog
 from intake.catalog.local import LocalCatalogEntry
 
-from . import __version__
+try:
+    from intake_stac import __version__
+except Exception:
+    __version__ = '999'
```
Comment (jhamman) [inline intake_stac/catalog.py:11]: @andersy005 - not sure why we aren't able to import the version here...

**Comment by jhamman** [comment]
per quick convo w/ @scottyhq - we're going to merge this as is, resolve conflicts with his #52 PR, and open dev up for others during the sprint.

**Pull request merged**